### PR TITLE
ROX-23308: Prefetch images for qa-e2e-tests

### PR DESCRIPTION
**Pull request by gavin-stackrox**
## Description

- This also changes the place where images are prefetched for the operator job
- There are two invocations:
   - one for images which already exist before a CI run starts - prefetching these can start ASAP
   - one for images which need to be built (asynchronously by GHA workflows) before they can be fetched. In order not to ruin the retry logic in prefetcher, it's best not to start prefetching them before we can confirm they exist on Quay

## Checklist
- [x] Investigated and inspected CI test results: the CI failure is unrelated
- [x] ~Unit test and regression tests added~
- [x] ~Evaluated and added CHANGELOG entry if required~
- [x] ~Determined and documented upgrade steps~
- [x] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

CI is sufficient.
- [x] 10x runs without new flakes
- [x] 10x with `all-qa` label without new flakes
  - added 2024-06-07 after 10ish runs
- [x] operator prefetch continues to function

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by porridge — APPROVED** [review]
LGTM now, @gavin-stackrox PTAL at my changes and merge if this LGTY too.

File: scripts/ci/lib.sh [hunk 1]
```diff
@@ -516,13 +516,60 @@ poll_for_system_test_images() {
...
+    # TODO(ROX-20508): for operaror-e2e jobs, pre-fetch images of the release from which operator upgrade test starts.
+    *)
+        info "No pre-built image prefetching is currently performed for: ${CI_JOB_NAME}"
+        ;;
+    esac
+}
+
+_image_prefetcher_system_start() {
+    case "$CI_JOB_NAME" in
+    *-operator-e2e-tests|gke*qa-e2e-tests)
```
Comment (porridge) [inline scripts/ci/lib.sh:549]: Actually I only noticed the GKE prefix now. Is there anything holding us from enabling it from other qa-e2e jobs as well?
```suggestion
    *-operator-e2e-tests|*qa-e2e-tests)
```

Comment (gavin-stackrox) [reply scripts/ci/lib.sh:549]: It would also be good for more coverage.

**Comment by gavin-stackrox** [comment]
> /retest

why no help bot?

**Comment by porridge** [comment]
@gavin-stackrox one thing I noticed when looking at this [failure](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/11248/pull-ci-stackrox-stackrox-master-gke-qa-e2e-tests/1798606266600263680) is that we don't wait for the `stackrox-images` prefetch to complete before deploying stackrox. This means pulls of some images might be happening concurrently from the prefetcher and from kubelet, slowing things down somewhat.

I don't know whether this is what caused this failure, as there are few timestamps around the SR deployment...

Ideally we'd:
1. start qa images prefetching
2. wait for images to be built
3. start stackrox images prefetching
4. wait for stackrox images prefetching to complete
5. deploy stackrox
6. wait for qa images prefetching to complete
7. start tests

**Comment by gavin-stackrox** [comment]
> @gavin-stackrox one thing I noticed when looking at this [failure](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/11248/pull-ci-stackrox-stackrox-master-gke-qa-e2e-tests/1798606266600263680) is that we don't wait for the `stackrox-images` prefetch to complete before deploying stackrox. This means pulls of some images might be happening concurrently from the prefetcher and from kubelet, slowing things down somewhat.
> 
> I don't know whether this is what caused this failure, as there are few timestamps around the SR deployment...
> 
> Ideally we'd:
> 
> 1. start qa images prefetching
> 2. wait for images to be built
> 3. start stackrox images prefetching
> 4. wait for stackrox images prefetching to complete
> 5. deploy stackrox
> 6. wait for qa images prefetching to complete
> 7. start tests

I think changing to this (1-7) logic should be easy enough. I'll TAL.

**Pull request merged**